packages/smithy-client/src/extensions/defaultClientConfiguration.ts

```diff
@@ -1,13 +1,8 @@
-import type { DefaultClientConfiguration } from "@smithy/types";
-
-import { getChecksumConfiguration, resolveChecksumRuntimeConfig } from "./checksum";
-
-/**
- * @internal
- */
-export type DefaultExtensionConfigType = Parameters<typeof getChecksumConfiguration>[0];
+import { getChecksumConfiguration } from "./checksum";
+import { DefaultExtensionConfigType } from "./defaultExtensionConfiguration";
 
 /**
+ * @deprecated will be replaced by getDefaultExtensionConfiguration
```

Comment:
move this as a stub into defaultExtensionConfiguration?

```ts
export const getDefaultClientConfiguration = getDefaultExtensionConfiguration;
```

Reply:
that way this file can be deleted. We maintain package API compatibility at the index, but not down to the filename level.